src/stash/src/lib.rs

```diff
@@ -238,8 +125,24 @@ where
     ///
     /// Intuitively, this method declares that all times less than `upper` are
     /// definite.
-    pub fn seal(&self, new_upper: AntichainRef<Timestamp>) -> Result<(), StashError> {
-        self.conn.seal(self.collection_id, new_upper)
+    fn seal<K, V>(
+        &self,
+        collection: StashCollection<K, V>,
+        upper: AntichainRef<Timestamp>,
+    ) -> Result<(), StashError>;
+
+    /// Performs multiple seals at once, potentially in a more performant way than
+    /// performing the individual seals one by one.
+    ///
+    /// See [Stash::seal]
+    fn seal_batch<K, V>(
+        &self,
+        seals: &[(StashCollection<K, V>, AntichainRef<Timestamp>)],
+    ) -> Result<(), StashError> {
+        for (id, new_upper) in seals {
+            self.seal(*id, *new_upper)?;
```

Comment:
This concerns me because it will make some seal implementations atomic and others not atomic. Proposal: Make the _batch functions required by the trait, and have the non-batch functions be the provided methods, and require the trait impl to ensure all batch operations are atomic.

Reply:
Another idea: merge this PR as is and I'll do this followup.

Reply:
IIUC the persist/storage ones won't be atomic, so I'd like to push back on making that the contract

Reply:
Hrm. Then we should say so in the docs and force callers to care.